stdlib/src/builtin/sort.mojo

```diff
@@ -15,400 +15,169 @@
+    @always_inline
+    @parameter
+    fn _partition(low: Int, high: Int) -> Int:
+        var pivot = list[high]
+        var i = low - 1
+        for j in range(low, high):
+            if list[j] <= pivot:
+                i += 1
+                list[j], list[i] = list[i], list[j]
+        list[i + 1], list[high] = list[high], list[i + 1]
+        return i + 1
+
+    if low < high:
+        var pi = _partition(low, high)
+        _quick_sort(list, low, pi - 1)
+        _quick_sort(list, pi + 1, high)
 
 
 @always_inline
-fn _sort2[
-    type: AnyRegType, cmp_fn: _cmp_fn_type
-](array: Pointer[type], offset0: Int, offset1: Int):
-    var a = array[offset0]
-    var b = array[offset1]
-    if not cmp_fn[type](a, b):
-        array[offset0] = b
-        array[offset1] = a
+fn quick_sort[D: ComparableCollectionElement](inout list: List[D]):
+    """Sort list of the order comparable elements in place with quick sort algorithm.
 
+    Parameters:
+        D: The order comparable collection element type.
 
-@always_inline
-fn _sort_partial_3[
-    type: AnyRegType, cmp_fn: _cmp_fn_type
-](array: Pointer[type], offset0: Int, offset1: Int, offset2: Int):
-    var a = array[offset0]
-    var b = array[offset1]
-    var c = array[offset2]
-    var r = cmp_fn[type](c, a)
-    var t = c if r else a
-    if r:
-        array[offset2] = a
-    if cmp_fn[type](b, t):
-        array[offset0] = b
-        array[offset1] = t
-    elif r:
-        array[offset0] = t
-
+    Args:
+        list: The list of the order comparable elements which will be sorted inpace.
+    """
+    _quick_sort(list, 0, len(list) - 1)
 
-@always_inline
-fn _small_sort[
-    n: Int, type: AnyRegType, cmp_fn: _cmp_fn_type
-](array: Pointer[type]):
-    @parameter
-    if n == 2:
-        _sort2[type, cmp_fn](array, 0, 1)
-        return
 
-    @parameter
-    if n == 3:
-        _sort2[type, cmp_fn](array, 1, 2)
-        _sort_partial_3[type, cmp_fn](array, 0, 1, 2)
-        return
+fn sort[D: ComparableCollectionElement](inout list: List[D]):
+    """Sort list of the order comparable elements in place. This function picks the best algorithm based on the list length.
 
-    @parameter
-    if n == 4:
-        _sort2[type, cmp_fn](array, 0, 2)
-        _sort2[type, cmp_fn](array, 1, 3)
-        _sort2[type, cmp_fn](array, 0, 1)
-        _sort2[type, cmp_fn](array, 2, 3)
-        _sort2[type, cmp_fn](array, 1, 2)
-        return
+    Parameters:
+        D: The order comparable collection element type.
 
-    @parameter
-    if n == 5:
-        _sort2[type, cmp_fn](array, 0, 1)
-        _sort2[type, cmp_fn](array, 3, 4)
-        _sort_partial_3[type, cmp_fn](array, 2, 3, 4)
-        _sort2[type, cmp_fn](array, 1, 4)
-        _sort_partial_3[type, cmp_fn](array, 0, 2, 3)
-        _sort_partial_3[type, cmp_fn](array, 1, 2, 3)
-        return
+    Args:
+        list: The list of the scalars which will be sorted inpace.
+    """
+    var count = len(list)
+    if count <= 64:
```

Comment:
**Question** From your experience, is this list size for the hybrid sort platform dependent or `64` is about the sweet spot across the board?